Update documentation with service accounts (#110)
1Password/terraform-provider-onepassword: docs/index.md

- `service_account_token` (String) A valid token for your 1Password Service Account. Can also be sourced from OP_SERVICE_ACCOUNT_TOKEN. Either this or `token` must be set.
- `token` (String) A valid token for your 1Password Connect API. Can also be sourced from OP_CONNECT_TOKEN. Either this or `service_account_token` must be set.

> - `url` (String) The HTTP(S) URL where your 1Password Connect API can be found. Must be provided through the OP_CONNECT_HOST environment variable if this attribute is not set. Can be omitted, if service_account_token is set.

Author: The thing is that this schema is generated automatically by `tfplugindocs`. Look like there is no way to put them on the table.

> - `url` (String) The HTTP(S) URL where your 1Password Connect API can be found. Must be provided through the OP_CONNECT_HOST environment variable if this attribute is not set. Can be omitted, if service_account_token is set.

Reviewer: Ah, no worries then 🙂

> - `url` (String) The HTTP(S) URL where your 1Password Connect API can be found. Must be provided through the OP_CONNECT_HOST environment variable if this attribute is not set. Can be omitted, if service_account_token is set.

Reviewer 2: All the files in the `docs` directory are generated. Therefore, there's not much we can change here.

## Known Service Accounts limitation:

> Users may encounter the following error `op error: (409) Conflict: Internal server conflict` when create/update/delete a bunch of items in the same vault as Terraform Provider handles each resource separately and therefore it makes a bunch of parallel requests using CLI for each of the resources.

Reviewer 2: `[nitpick]`

Some phrasing improvements:
```suggestion
Users may encounter the following error `op error: (409) Conflict: Internal server conflict` when creating/updating/deleting multiple items in the same vault as Terraform handles each resource separately and therefore it makes parallel requests using the CLI for each resource.
```

Author: Thank you for the suggestion! I like the way @armstrongl represents it 😃

There are ways to avoid this:
1. Use `depends_on` in your resource definition to make sure the Provider makes requests sequentially.

> 2. After it fails with `409` error run `terraform apply` again till all the changes will be applied.

Reviewer 2: `[nitpick]`

```suggestion
2. After it fails with `409` error, run `terraform apply` again till all the changes will be applied.
```

3. Use Connect.

> 4. Put items in the different vaults.

Reviewer: I might be misunderstanding, but my initial reaction to this is that the user probably has a 1Password account already set up, and it wouldn't make sense to reorganize the items into different vaults just to avoid this error. Also, where do we draw the line? Is there a target number of items per vault that would avoid this issue?

Author: Good question. Race condition errors may occur when there are 2 or more items involved. In practice, most of the time it happens when there are 3 or more items.

`Put items in the different vaults.` - this is for the case when you want to create new items using terraform configuration.

There is an open issue to fix that #108. So, this section is temporary here.
